zerver/lib/mention.py

```diff
@@ -74,6 +87,8 @@ def __init__(self, realm_id: int) -> None:
         self.realm_id = realm_id
         self.user_cache: Dict[Tuple[int, str], FullNameInfo] = {}
         self.stream_cache: Dict[str, int] = {}
+        self.recipient_cache: Dict[str, ChannelInfo] = {}
```

Comment:
Do we need `recipient_cache`? It seems like you could instead just change the type of `stream_cache` to be a dataclass with a few fields, not just the channel ID, rather than adding a new data structure.

Reply:
ya, I guess that would also be fine, I'll update it